documentation/projects/proposals/rekognition_data/20240510-implementation_plan_machine_generated_tags_frontend.md

```diff
@@ -0,0 +1,199 @@
+be linked from the Generated tags section.
+
+## Dependencies
+
+This project needs the API to add the `unstable__provider` property to tags in
+media responses, which was implemented in
+[Expose provider in the API tags response #4280](https://github.com/WordPress/openverse/pull/4280)
+
+### Feature flags
+
+<!-- List feature flags/environment variables that will be utilised in the development of this plan. -->
+
+The project will contain at most 2 PRs that can be reverted, so the feature
+flags are not necessary.
+
+### Infrastructure
+
+<!-- Describe any infrastructure that will need to be provisioned or modified. In particular, identify associated potential cost changes. -->
+
+No infrastructure changes will be necessary, we will only update the single
+result pages of the Nuxt frontend.
+
+## Alternatives
+
+<!-- Describe any alternatives considered and why they were not chosen or recommended. -->
+
+The tags will be displayed in two sections: "Source tags" and "Generated tags".
+
+For the machine-generated tags, we want to add a link to the page that describes
+We have the provider (currently, it will be "clarifai" and "rekognition") and
+the accuracy data for the machine-generated providers.
+
+There are several options of displaying these properties.
+
+1. Do not display this data. The two tags sections will be identical.
+2. The generated tags will have a tooltip that displays the tag's provider and
+   its accuracy.
+3. Display generated tags in sections by provider so that we don't duplicate the
+   tag provider on each tag. Then, on each tag, display the accuracy.
+
+   The first option is the simplest to implement, however, it fails to provide
+   the fullest information to the users. The second option provides the accuracy
+   and provider information to the user, however it is not very accessible.
+   Different browsers and screen readers treat the tooltips differently. We have
+   enough room on the page to display this data in full, using the third option:
+   labelling all the tags generated by a single provider with a "by provider"
+   heading, and adding the accuracy to the tag's inner content. This way, the
+   user will see the provider and the accuracy of the tag without any additional
+   actions. As a bonus, if a tag is generated by multiple providers, the user
+   will see that different providers generate the same tag, and will be able to
+   see the difference in accuracy between the providers.
+
+## Design
+
+<!-- Note any design requirements for this plan. -->
+
+The first iteration of the designs that use the first alternative option is
+ready:
+[Displaying machine-generated content](https://github.com/WordPress/openverse/issues/4192)
+If we decide to go with the third alternative option, we will need to add the
+provider headings and the accuracy to the tags.
+
+## Blockers
+
+<!-- What hard blockers exist that prevent further work on this project? -->
+
+There are no blockers for this work.
```

Comment:
A little more context here might be helpful:
```suggestion
Since we already have machine-generated tags available in the API, there are no blockers for this work.
```